packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -80,6 +80,78 @@ void main() {
     debugDefaultTargetPlatformOverride = null;
   });
 
+  testWidgets('Using other pickers that rebuild the switch will not cause vibrations', (WidgetTester tester) async {
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+    final Key switchKey = UniqueKey();
+    final Key switchKey2 = UniqueKey();
+    bool value = false;
+    bool value2 = false;
+    final List<MethodCall> log = <MethodCall>[];
+
+    SystemChannels.platform.setMockMethodCallHandler((MethodCall methodCall) async {
+      log.add(methodCall);
+    });
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Center(
+              child: Column(
+                children: <Widget>[
+                  CupertinoSwitch(
+                    key: switchKey,
+                    value: value,
+                    onChanged: (bool newValue) {
+                      setState(() {
+                        value = newValue;
+                      });
+                    },
+                  ),
+                  CupertinoSwitch(
+                    key: switchKey2,
+                    value: value2,
+                    onChanged: (bool newValue) {
+                      setState(() {
+                        value2 = newValue;
+                      });
+                    },
+                  ),
+                ]
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    await tester.tap(find.byKey(switchKey));
+    await tester.pumpAndSettle();
```

Comment:
In general, we want to avoid pumpAndSettle and be specific in the timing of the assertion. For instance, here, we're just checking that a haptic kick happened at some point. But we should more specifically test that the timing of the knock coincides with the timing of the knob reaching the edge (or whatever iOS does).